### PR TITLE
Add intel backend (triton gpu to llvm convertion)

### PR DESCRIPTION
**Pull request by kurapov-peter**
This is the first step to move our flow to the backend. Since the first two attempts failed miserably this does the easiest thing possible - move the existing pipeline to the backend resolving all the name collisions.

Since we use nvgpu&friends in our pipeline, I've created a copy of those in the backend. This proved to be necessary when merging upstream changes. We will keep it there and gradually clean up.

I kept namespace changes to a minimum (some of the resolution is done by simple renaming) partly because of potential ambiguity and partly because most of it will go away with the next steps (mostly related to utilities which will be split into generic ones in lib/conversion and backend-specific).

The tests now still run through the original pipeline (I'm being cautious). I suggest we switch the execution to the new flow right after this and then proceed with the merge/cleanup.

### CODE REVIEW

**Comment by whitneywhtsang** [comment]
To do it similar to amd, should we use namespace like `using ::AMD::TritonGPUToLLVMTypeConverter;` instead of renaming to `TritonIntelGPUToLLVMTypeConverter` for `TritonGPUToLLVMTypeConverter`?

File: third_party/intel/lib/TritonIntelGPUToLLVM/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    Utility.cpp
+    ViewOpToLLVM.cpp
+    TensorPtrOpsToLLVM.cpp
+    ClusterOpsToLLVM.cpp
+    PTXAsmFormat.cpp
+    AllocateSharedMemory.cpp
+
+    DEPENDS
+    TritonIntelGPUConversionPassIncGen
+    NVGPUIntelConversionPassIncGen
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUToLLVM/CMakeLists.txt:37]: why  `TritonGPUToLLVM` doesn't dpends on `NVGPUConversionPassIncGen`, but `TritonIntelGPUToLLVM` dpends on `NVGPUIntelConversionPassIncGen` here?

**Comment by kurapov-peter** [comment]
> To do it similar to amd, should we use namespace like `using ::AMD::TritonGPUToLLVMTypeConverter;` instead of renaming to `TritonIntelGPUToLLVMTypeConverter` for `TritonGPUToLLVMTypeConverter`?

This is just more typing to do the same thing. I expect us to remove the converter altogether anyway.

**Comment by kurapov-peter** [comment]
> The tests now still run through the original pipeline (I'm being cautious). I suggest we switch the execution to the new flow right after this and then proceed with the merge/cleanup.

Just checked. I wasn't cautious enough, they actually run through the new path :)

**Comment by whitneywhtsang** [comment]
> > To do it similar to amd, should we use namespace like `using ::AMD::TritonGPUToLLVMTypeConverter;` instead of renaming to `TritonIntelGPUToLLVMTypeConverter` for `TritonGPUToLLVMTypeConverter`?
> 
> This is just more typing to do the same thing. I expect us to remove the converter altogether anyway.

That's fine, let's revisit after the merge is completed if needed.

**Review by whitneywhtsang — APPROVED** [review]

**Review by ienkovich — APPROVED** [review]

**Review by etiotto — APPROVED** [review]

**Pull request merged**